addon/model-data.js

```diff
@@ -246,11 +213,7 @@ export default class M3ModelData {
     if (this.isDestroyed) {
       return;
     }
-    this.destroy();
```

Comment:
I've renamed the function to be `_destroy()` instead of `destroy()` to mark it as something, which externally no one should call (should always use `unloadRecord()`). However, I might be missing some nuances of what `isDestroyed` and `destroy()` do.